common/namespace/handler.go

```diff
@@ -795,6 +815,52 @@ func (d *HandlerImpl) validateVisibilityArchivalURI(URIString string) error {
 	return archiver.ValidateURI(URI)
 }
 
+// maybeUpdateFailoverHistory adds an entry if the Namespace is becoming active in a new cluster.
+func (d *HandlerImpl) maybeUpdateFailoverHistory(
+	failoverHistory []*persistencespb.FailoverStatus,
+	updateReplicationConfig *replicationpb.NamespaceReplicationConfig,
+	namespaceDetail *persistencespb.NamespaceDetail,
+	newFailoverVersion int64,
+) []*persistencespb.FailoverStatus {
+	d.logger.Debug(
+		"maybeUpdateFailoverHistory",
+		tag.NewAnyTag("failoverHistory", failoverHistory),
+		tag.NewAnyTag("updateReplConfig", updateReplicationConfig),
+		tag.NewAnyTag("namespaceDetail", namespaceDetail),
+	)
+	if updateReplicationConfig == nil {
+		d.logger.Debug("updateReplicationConfig was nil")
+		return failoverHistory
+	}
+	desiredReplicationState := updateReplicationConfig.State
+	if desiredReplicationState == enumspb.REPLICATION_STATE_UNSPECIFIED {
+		desiredReplicationState = namespaceDetail.ReplicationConfig.State
+	}
+	// N.B., UNSPECIFIED is the same as NORMAL
+	if desiredReplicationState != enumspb.REPLICATION_STATE_NORMAL &&
+		desiredReplicationState != enumspb.REPLICATION_STATE_UNSPECIFIED {
+		d.logger.Debug("Replication state not NORMAL", tag.NewAnyTag("state", updateReplicationConfig.State))
+		return failoverHistory
+	}
+	lastFailoverVersion := int64(-1)
+	if l := len(namespaceDetail.ReplicationConfig.FailoverHistory); l > 0 {
+		lastFailoverVersion = namespaceDetail.ReplicationConfig.FailoverHistory[l-1].FailoverVersion
+	}
+	if lastFailoverVersion != newFailoverVersion {
+		now := d.timeSource.Now()
+		failoverHistory = append(
+			failoverHistory, &persistencespb.FailoverStatus{
+				FailoverTime:    &now,
+				FailoverVersion: newFailoverVersion,
+			},
+		)
+	}
+	if l := len(failoverHistory); l > maxReplicationHistorySize {
+		failoverHistory = failoverHistory[l-maxReplicationHistorySize : l]
```

Comment:
is this the same as failoverHistory[1:]?

Reply:
No, it is taking the last `maxReplicationHistorySize` elements.